hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -572,8 +595,39 @@ private static boolean hiveEngineEnabled(TableMetadata metadata, Configuration c
         ConfigProperties.ENGINE_HIVE_ENABLED, TableProperties.ENGINE_HIVE_ENABLED_DEFAULT);
   }
 
+  /**
+   * Returns if the hive locking should be enabled on the table, or not.
+   *
+   * <p>The decision is made like this:
+   *
+   * <ol>
+   *   <li>Table property value {@link TableProperties#HIVE_LOCK_ENABLED}
+   *   <li>If the table property is not set then check the hive-site.xml property value {@link
+   *       ConfigProperties#LOCK_HIVE_ENABLED}
+   *   <li>If none of the above is enabled then use the default value {@link
+   *       TableProperties#HIVE_LOCK_ENABLED_DEFAULT}
+   * </ol>
+   *
+   * @param metadata Table metadata to use
+   * @param conf The hive configuration to use
+   * @return if the hive engine related values should be enabled or not
+   */
+  private static boolean hiveLockEnabled(TableMetadata metadata, Configuration conf) {
```

Comment:
Also, is there any way to validate Hive compaitibility, to prevent old Hive version from disabling Hive locks?

Reply:
Since there is no official release with the new code yet, sadly this is not an option 😢

Reply:
General question: is there any way we can check Hive server version?  Would be nice in general.

Reply:
It is.
So once it get released, we can check the version.

Reply:
Great, I think this would be very important to get in for users of OSS Hive, its a bit hacky otherwise.

Reply:
Let's make an issue to track it on Iceberg side

Reply:
Created #7418